tokio/src/sync/mpsc/bounded.rs

```diff
@@ -230,6 +230,38 @@ impl<T> Receiver<T> {
         poll_fn(|cx| self.chan.recv(cx)).await
     }
 
+    /// Receives the next values for this receiver and populates `buffer`.
+    ///
+    /// This method returns the number of values populated in `buffer`,
+    /// which is cleared on each call.
+    ///
+    /// If the buffer initially has zero capacity, the method reserves
+    /// `BLOCK_CAP` elements, otherwise the capacity is not changed.
+    /// The number of returned values can be at most the capacity of
+    /// the buffer.
+    ///
+    /// # Example
+    /// ```
+    /// use tokio::sync::mpsc;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let (tx, mut rx) = mpsc::channel(100);
+    ///
+    ///     tx.send("hello").await.unwrap();
+    ///     tx.send("world").await.unwrap();
+    ///
+    ///     // At most 5 at a time
+    ///     let mut buffer : Vec<&str> = Vec::with_capacity(5);
+    ///     assert_eq!(2, rx.recv_many(&mut buffer).await);
+    ///     assert_eq!(vec!["hello","world"], buffer);
+    /// }
+    /// ```
+    pub async fn recv_many(&mut self, buffer: &mut Vec<T>) -> usize {
```

Comment:
I would document that this method only returns zero if the channel is closed.

Reply:
documentation was updated to reflect this.